streams/src/main/java/org/apache/kafka/streams/errors/StreamsException.java

```diff
@@ -57,19 +73,29 @@ public StreamsException(final Throwable throwable) {
 
     public StreamsException(final Throwable throwable, final TaskId taskId) {
         super(throwable);
-        this.taskId = taskId;
+        this.taskIds.add(taskId);
     }
 
     /**
      * @return  the {@link TaskId} that this exception originated from, or {@link Optional#empty()} if the exception
      *          cannot be traced back to a particular task. Note that the {@code TaskId} being empty does not
      *          guarantee that the exception wasn't directly related to a specific task.
      */
+    @Deprecated
```

Comment:
Why do we need to deprecate this method? (Same question below).

Reply:
I thought it would be better to generalise `StreamsException` and allow it to hold multiple taskIDs. In that case you would not need a getter for a single task ID.